apache2/waf_logging/waf_log_util.h

```diff
@@ -0,0 +1,27 @@
+#ifndef _WAF_LOG_UTIL_EXTERNAL_HEADER
+#define _WAF_LOG_UTIL_EXTERNAL_HEADER
+
+#define WAF_LOG_UTIL_FAILED -1
+#define WAF_LOG_UTIL_SUCCESS 0
+#define MODSEC_MODE_DETECT 1
+#define MODSEC_MODE_PREVENT 2
+#define WAF_RULESET_PREFIX "/RuleSets/"
+#define WAF_LOG_UTIL_FILE "waf_json.log"
+#define WAF_LOG_UTIL_OPERATION_NAME "ApplicationGatewayFirewall"
+#define WAF_LOG_UTIL_CATEGORY "ApplicationGatewayFirewallLog"
+
+#define RULE_TYPE_OWASP_CRS "OWASP_CRS"
+
+#define RULE_HASH_SIZE 499
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+char* generate_json(const char* resourceId, const char* operationName, const char* category, const char* instanceId, const char* clientIP, const char* clientPort, const char* requestUri, const char* ruleSetType, const char* ruleSetVersion, const char* ruleId, const char* messages, const int action, const int site, const char* details_messages, const char* details_data, const char* details_file, const char* details_line, const char* hostname, const char* waf_unique_id);
```

Comment:
What is waf_unique_id? And should it be changing per log message?

Reply:
This is what is reflected in logs as `transactionId`. I don't know much about this specific parameter, maybe @alonzop or @azhao155 can shed some light on it. But my changes don't alter anything in regards to how this parameter is handled so honestly I did not really bother to figure it out :)